### PR TITLE
housekeeping: add OpenAPI document and tooling

### PR DESCRIPTION
**Pull request by shockey**
This PR:

* adds a `/docs` route to the API, which renders our OpenAPI document with Swagger UI
* adds an `openapi:edit` npm script, which launches Swagger Editor locally and syncs changes back to the filesystem
* introduces an initial OpenAPI document for the API

### CODE REVIEW

**Comment by spraguejdev** [comment]
Awesome! Thanks you for the help Kyle. As soon as everyone else looks at it, let's merge the PR.

File: server/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      responses:
+        200:
+          description: OK; data in response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Channel'
+  /channels/{channelId}:
```
Comment (spraguejdev) [inline server/openapi.yaml]: Is this endpoint necessary? Like what will /channels/:channelId return? Just a thought. Seems like /channels/{channelId}/messages will return all the messages for a specific channel. What do you think?

File: server/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                example: contractor-description.pdf
+              fileType:
+                type: string
+                example: pdf
+              downloadUrl:
+                type: string
+                format: url
+                example: "https://hrxp.com/static/files/FJ0UW704E/contractor-description.pdf"
+        replies:
+          type: array
```
Comment (spraguejdev) [inline server/openapi.yaml:90]: I think we are missing a text field here?

File: server/openapi.yaml (hunk 1 above)
Comment (shockey) [reply server/openapi.yaml]: > ~is this endpoint necessary?~

~nope, I don't think there are any details about messages that we won't expose via `/channels/{channelId}/messages`. Let's remove it.~

Comment (shockey) [reply server/openapi.yaml]: Oops! Thought this was `/channels/{channelId}/messages/{messageId}`. I removed that in eae4f0cece580, LMK if you disagree with that.

As for `/channels/{channelId}`, I could imagine it being useful for getting details about a specific channel, but there's no reason we could just have the frontend reuse the data from `/channels`. Let's remove it as well.

Comment (shockey) [reply server/openapi.yaml]: Resolved in 19d76c2.

**Review by miken619 — APPROVED** [review]

**Review by spraguejdev — APPROVED** [review]

File: server/openapi.yaml (hunk 2 above)
Comment (shockey) [reply server/openapi.yaml:90]: For sure, I'll fix that. 

We're also not tracking which channel a message belongs to, but if you're okay with it I'll leave that for now, and we can address that in #6 and have the changes affect both the model and the OpenAPI schema.

**Pull request merged**